components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js

```diff
@@ -1,264 +1,401 @@
+    onKeyPress : function($super, e) {
+      return $super(e, /[0-9]/, 2, true);
     },
 
-    monthSelected : function(doNotNotifyOnChange) {
-      if (this.inputFormat == "Y") {
-          this.monthSelector.selectNone();
+    refreshAvailableValues : function() {
+      var availableValues = [];
+      for (var i = 1; i <= 12; i++) {
+        availableValues.push(("00" + i).slice(-2));
       }
-      if (this.monthSelector.getSelectedValue() > 0) {
-        this.daySelector.populate(this.getAvailableDays());
-        this.daySelector.enable();
-      } else {
-        if (this.inputFormat == "DMY") {
-            // in "DMY" mode let user pick any day if no month is selected
-            this.daySelector.populate(this.getZeroPaddedValueRange(1,31));
-        } else {
-            this.daySelector.disable();
-        }
+      this.availableValues = availableValues;
+      return availableValues;
+    },
+    formatValue : function(value) {
+      if (typeof value == 'number') {
+        value = value.toString();
       }
-      if (this.inputFormat == "MY" || this.inputFormat == "MMY") {
-          this.daySelector.selectNone();
+      if (value.length == 1) {
+        value = "0" + value;
       }
-      this.updateDate(doNotNotifyOnChange);
+      return value;
+    }
+  });
+
+  /**
+  * FuzzyDayInput is the text input day component of the @class FuzzyDatePicker
+  *
+  * @class FuzzyDayInput
+  * @Extends AbstractFuzzyDateInput
+  * @constructor
+  * @param {element} A text input element. The behaviour of the element will be modified.
+  * @param {alwaysEnabled}  Booolean indicating whether the day value should always be enabled. If false
+  *                         the value will be enabled iff the year input is valid and not fuzzy.
+  * @param {yearInput} The @class FuzzyYearInput to which this day input is associated with.
+  * @param {monthInput} The @class FuzzyMonthInput to which this day input is associated with.
+  */
+  var FuzzyDayInput = Class.create(AbstractFuzzyDateInput, {
+    initialize : function($super, element, alwaysEnabled, yearInput, monthInput) {
+      this.yearInput = yearInput;
+      this.monthInput = monthInput;
+      $super(element, alwaysEnabled);
     },
 
-    createDayDropdown : function(hide) {
-      this.daySelector = new PedigreeFuzzyDatePickerDropdown({name: "day", alwaysEnabled: (this.inputFormat != "YMD")});
-      this.daySelector.populate(this.getZeroPaddedValueRange(1,31));
-      this.daySelector.disable();
-      this.daySelector.onSelect(this.updateDate.bind(this));
-      if (hide) {
-          this.daySelector.getElement().hide();
-      }
-      return this.daySelector.getElement();
+    onKeyPress : function($super, e) {
+      return $super(e, /[0-9]/, 2, true);
     },
 
-    getAvailableDays : function () {
-      var year = this.yearSelector.getSelectedValue() * 1;
-      var month = this.monthSelector.getSelectedValue() * 1;
+    onDateChange : function($super) {
+      $super();
```

Comment:
We need to make sure this plays well with different date formats. I see the logic to make it `alwaysEnabled`, but somehow this is turned off when `yearInput.isFuzzy`. Maybe that is ok, need to think about that.